did-sdk/src/key/key.ts

```diff
@@ -0,0 +1,120 @@
+import * as key from '@transmute/did-key.js'
+import { DIDDocument } from 'did-resolver'
+import { ReadOnlyDidDocAPI, WritableDidDocAPI } from '../did-documents.js'
+
+export type generateFromSeedOptions = { secureRandom: () => Buffer }
+export type generateFromRandomOptions = {
+  kty: 'EC' | 'OKP'
+  crvOrSize: 'P-256' | 'P-384' | 'P-521'
+}
+export type KeyType =
```

Comment:
do we want to support NIST curves here? I always go back & forth on it. but webcrypto & a lot of device TPMs only support RSA + NIST curves. 

I'm using NIST P-256 in AWAKE for instance

Reply:
I decided to see what the deal is with webcrypto curves and came across [this interesting exchange](https://news.ycombinator.com/item?id=19749019). I know tptacek isn't the end-authority on all crypto but he tends to know what he's talking about. The other thing I found is unconfirmed statements that Microsoft has been the blocker on 25519.

We might not really have a choice *not* to use the NIST curves. We need the widest support possible (TPMs) and it's probably safer to be using WebCrypto than js/wasm versions of crypto.

Reply:
that's sorta my sense as well. TPM with NIST P-256 is definitely better than browser storage with ed25519. So we certainly shouldn't be dogmatic about not using NIST curves cause then we lose out on most available TPMs.

in the browser, the benefit of webcrypto is:
- you can store keys as "unexportable" in IDB. so an attack can do operations with them but not actually steal the key material
- operations are faster cause they're at the system level instead of javascript runtime which means you get some hardware acceleration on things like sha256

downside is obviously the downside of NIST over Ed25519:
- longer public keys
- not constant time
- possibility of a backdoor